### PR TITLE
Store Orders: Add "Resend Invoice" functionality to pending-payment orders

### PR DESCRIPTION
**Pull request by ryelle**
See #20794 – This PR gets the state & data requests up for manually sending an invoice email via the new API endpoint in https://github.com/Automattic/wc-calypso-bridge/pull/21. This also adds a "Resend Invoice" button to the header of all pending-payment orders:

resend-button

Clicking this triggers the API and should send off an invoice (once the updated wc-calypso-bridge is pushed, or if you've got master running on your site - otherwise it'll just error-notice at you)

There's a new action, reducer subtree, and selector for tracking the status of the invoice request; along with a new data layer handler to send the request.

**To test**

- Run the tests `npm run test-client client/extensions/woocommerce/state`
- Check out the button on an order -  `/store/orders/:site`, and pick an order pending payment
- You should see the Resend Invoice button in the top bar
- Clicking it will trigger an action, and if you have the updated bridge plugin, send an email invoice.
- The Activity Log will also be updated immediately with the note that you've sent the invoice. 

@kellychoffman @jameskoster Let me know if you meant a different location by "'Resend invoice' button appears in header." (also, I'll have a separate PR for adding the button to the success notice)

### CODE REVIEW

File: client/extensions/woocommerce/state/data-layer/orders/send-invoice/index.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import {
+	orderInvoiceFailure,
+	orderInvoiceSuccess,
+} from 'woocommerce/state/sites/orders/send-invoice/actions';
+import request from 'woocommerce/state/sites/http-request';
+import { WOOCOMMERCE_ORDER_INVOICE_SEND } from 'woocommerce/state/action-types';
+
+export const fetch = action => {
+	const { siteId, orderId } = action;
+	return request( siteId, action ).post( `orders/${ orderId }/send_invoice`, {} );
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/data-layer/orders/send-invoice/index.js]: Minor, but I think that `{}` will get passed by default for post requests, so you can leave this off.

File: client/extensions/woocommerce/state/data-layer/orders/send-invoice/test/index.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * Internal dependencies
+ */
+import { sendOrderInvoice } from 'woocommerce/state/sites/orders/send-invoice/actions';
+import { fetch } from '../';
+import { http } from 'state/data-layer/wpcom-http/actions';
+
+describe( 'handlers', () => {
+	describe( '#fetch', () => {
+		test( 'should dispatch a get action to the API via the jetpack proxy for this siteId & orderId', () => {
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/data-layer/orders/send-invoice/test/index.js]: `should dispatch a get action` --> `should dispatch a post action`

File: client/extensions/woocommerce/state/data-layer/orders/send-invoice/index.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { createNoteSuccess } from 'woocommerce/state/sites/orders/notes/actions';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import {
+	orderInvoiceFailure,
+	orderInvoiceSuccess,
+} from 'woocommerce/state/sites/orders/send-invoice/actions';
+import request from 'woocommerce/state/sites/http-request';
+import { WOOCOMMERCE_ORDER_INVOICE_SEND } from 'woocommerce/state/action-types';
+
+export const fetch = action => {
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/data-layer/orders/send-invoice/index.js]: Send might make more sense then `fetch` here.

Comment (ryelle) [reply client/extensions/woocommerce/state/data-layer/orders/send-invoice/index.js]: Though I agree, `fetch` is the property expected by `dispatchRequestEx`, and shouldn't be used outside of this file & the test, so I think I'm going to leave it as-is.

File: client/extensions/woocommerce/app/order/index.js [hunk 4]
```diff
@@ -134,11 +152,23 @@ class Order extends Component {
 			</Button>,
 		];
 		if ( ! isEditing ) {
-			button = (
-				<Button primary onClick={ this.toggleEditing }>
+			button = [
+				<Button key="edit" primary onClick={ this.toggleEditing }>
 					{ translate( 'Edit Order' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/order/index.js]: Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Edit Forum Order' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**